components/shared/button/Button.test.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import Button, { ButtonType } from './Button';
+
+describe('Button tests', () => {
+  it('No props, empty button', () => {
+    const { queryByTestId } = render(<Button></Button>);
+    const button = queryByTestId('button');
+    expect(button?.className).toContain('bg-white text-gray-700');
+    expect(button?.innerHTML).toBe('');
+  });
+  it('No props with children', () => {
+    const { queryByTestId } = render(<Button>Child element</Button>);
+    const button = queryByTestId('button');
+    expect(button?.className).toContain('bg-white text-gray-700');
+    expect(button?.innerHTML).toEqual('Child element');
+  });
+  it('type Black', () => {
+    const { queryByTestId } = render(<Button type={ButtonType.Black}></Button>);
+    const button = queryByTestId('button');
```

Comment:
za każdym razem definiujez tutaj button,
napisz jakąś funkcję wrapper na `render(<Button type={ButtonType.Black}></Button>)` która będzie przyjmowała jakieś propsy (np. type) i zwracała już gotowy button:
```js
it('test', () => {
const { button } = setup(children, type)
expect(button).toBeNull()
```
Zresztą jakim cudem dla pustych childrenów wyświetla się Button? chyba jest jakis bug w komponenncie.
to pewnie wynika z tego Twojego `button?`
TS na to nie powiinen pozwolić

Reply:
poprawione - children przerzucony do
`interface ButtonProps {
  children: string;
  type?: ButtonType;
}`